src/DynamoCore/Models/DynamoModelEventArgs.cs

```diff
@@ -172,6 +176,12 @@ public EvaluationCompletedEventArgs(bool evaluationTookPlace, Exception errorMsg
 
             error = errorMsg != null ? Option.Some(errorMsg) : Option.None<Exception>();
         }
+        internal EvaluationCompletedEventArgs(bool evaluationTookPlace, IEnumerable<string> messageKeys, Exception errorMsg = null)
+        {
+            EvaluationTookPlace = evaluationTookPlace;
+            MessageKeys = messageKeys;
+            error = errorMsg != null ? Option.Some(errorMsg) : Option.None<Exception>();
+        }
     }
```

Comment:
Does `public EvaluationCompletedEventArgs(bool evaluationTookPlace, Exception errorMsg = null)` need to be deprecated as a result of adding this new constructor?

Reply:
I don't think we *need* though we could, I've made the new constructor internal for now, I think we should initialize the `MessageKeys` to an empty list probably though.